### ISSUE TITLE
hypercorn not working on Windows 10 ?

### ISSUE THREAD

**Issue by stonebig** [pgjones/hypercorn#23]
hi,

trying on Python-3.7 and 3.8rc the "hello" example, I fall on "add_signal_handler" not implemented in asyncio. is it Normal ?

I had hopes hypercorn was ok under windows, so I may have done something wrong ?

````
  File "C:\WinP\bd38\bucod\WPy64-3800rc1\python-3.8.0rc1.amd64\lib\site-packages\hypercorn\asyncio\run.py", line 212, in _run
    loop.add_signal_handler(signal.SIGINT, _signal_handler)
  File "C:\WinP\bd38\bucod\WPy64-3800rc1\python-3.8.0rc1.amd64\lib\asyncio\events.py", line 536, in add_signal_handler
    raise NotImplementedError
NotImplementedError
````

hello.py
````python3
async def app(scope, receive, send):
    if scope["type"] != "http":
        raise Exception("Only the HTTP protocol is supported")

    await send({
        'type': 'http.response.start',
        'status': 200,
        'headers': [
            (b'content-type', b'text/plain'),
            (b'content-length', b'5'),
        ],
    })
    await send({
        'type': 'http.response.body',
        'body': b'hello',
    })
````

**Comment by stonebig** [comment]
hum, I see some maybe hope there, but not expert in the thing:
https://github.com/encode/uvicorn/pull/423

**Comment by pgjones** [comment]
Hmm, this used to be an `AttributeError`, I think Python 3.8 has changed the affected error. Fixed in a8ba7f62dd8f1a937ecf1ca3e5486e7d64df3e54. You should be able to use Hypercorn with Python 3.7 without this issue.

**Comment by stonebig** [comment]
maybe for some under Anaconda, it doesn't work either on my Python-3.7. 
... I'll wait to see uvicorn/hypercorn next releases, I hoped it was something obvious.

````
C:\WinP\bd37\bucod\WPy64-3750b1\notebooks>hypercorn hello:app
Traceback (most recent call last):
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\Scripts\hypercorn.exe\__main__.py", line 9, in <module>
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\site-packages\hypercorn\__main__.py", line 229, in main
    run(config)
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\site-packages\hypercorn\run.py", line 34, in run
    worker_func(config)
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\site-packages\hypercorn\asyncio\run.py", line 167, in asyncio_worker
    shutdown_trigger=shutdown_trigger,
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\site-packages\hypercorn\asyncio\run.py", line 212, in _run
    loop.add_signal_handler(signal.SIGINT, _signal_handler)
  File "C:\WinP\bd37\bucod\WPy64-3750b1\python-3.7.5rc1.amd64\lib\asyncio\events.py", line 540, in add_signal_handler
    raise NotImplementedError
NotImplementedError

C:\WinP\bd37\bucod\WPy64-3750b1\notebooks>
````
pip list attached, in case there is a subtle detail there 
[toto.txt](https://github.com/pgjones/hypercorn/files/3721001/toto.txt)

**Comment by stonebig** [comment]
so I can get today uvicorn working on Python-3.7 and Python-3.8. I don't catch what detail I have  wrong with hypercorn.

**Comment by stonebig** [comment]
well, with hypercorn-0.9.0dev master it works now on Python-3.8.

**Comment by pgjones** [comment]
Great, thanks for confirming.

**Comment by ghostzero192** [comment]
having this exact problem with python 3.7.5 x64, any idea how to fix this?

**Comment by pgjones** [comment]
@ghostzero192 If you make use of the master branch it will work, specifically a8ba7f6.

**Comment by ghostzero192** [comment]
thank you so much, it works perfectly!!

**Comment by ghostzero192** [comment]
using python 3.8 on windows i get this error

`hypercorn main:app -w 2`

`Running on 127.0.0.1:8000 over http (CTRL + C to quit)
Process Process-1:
Traceback (most recent call last):
  File "c:\Program Files\Python38\lib\multiprocessing\process.py", line 313, in _bootstrap
    self.run()
  File "c:\Program Files\Python38\lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "c:\program files\python38\testenv\lib\site-packages\hypercorn\asyncio\run.py", line 164, in asyncio_worker
    _run(
  File "c:\program files\python38\testenv\lib\site-packages\hypercorn\asyncio\run.py", line 220, in _run
    loop.run_until_complete(main(shutdown_trigger=shutdown_trigger))
  File "c:\Program Files\Python38\lib\asyncio\base_events.py", line 608, in run_until_complete
    return future.result()
  File "c:\program files\python38\testenv\lib\site-packages\hypercorn\asyncio\run.py", line 114, in worker_serve
    await asyncio.start_server(
  File "c:\Program Files\Python38\lib\asyncio\streams.py", line 94, in start_server
    return await loop.create_server(factory, host, port, **kwds)
  File "c:\Program Files\Python38\lib\asyncio\base_events.py", line 1469, in create_server
    server._start_serving()
  File "c:\Program Files\Python38\lib\asyncio\base_events.py", line 309, in _start_serving
    sock.listen(self._backlog)
OSError: [WinError 10022] An invalid argument was supplied`

**Comment by pgjones** [comment]
@ghostzero192 Is it just Python 3.7 -> 3.8 that changed (between the working and not working versions)?

**Comment by ghostzero192** [comment]
@pgjones 
in python 3.7 everything works fine if i use the branch that you recomended me (0.9.0+dev), but in python 3.8 neither the master branch or the 0.9.0+dev work

**Comment by pgjones** [comment]
@ghostzero192 I think this is fixed with 0.9.2, see 42c2f6f5c9534f4a8912614a8a19a6317ee01a26
